packages/node-cli/src/commands/actionresult/CreateProjectActionResult.js

```diff
@@ -0,0 +1,109 @@
+/*
+** Copyright (c) 2019 Oracle and/or its affiliates.  All rights reserved.
+** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+*/
+'use strict';
+const assert = require('assert');
+const ActionResult = require('./ActionResult');
+
+class CreateProjectActionResult extends ActionResult {
+
+	constructor(build) {
+		super(build)
+		this._projectType = build.projectType;
+		this._projectDirectory = build.projectDirectory;
+		this._includeUnitTesting = build.includeUnitTesting;
+		this._npmInstallSuccess = build.npmInstallSuccess;
+	}
+
+	validateBuild(build) {
+		assert(build);
+		assert(build.status, "status is required when creating an ActionResult object.");
+		if (build.status === ActionResult.SUCCESS) {
+			assert(build.data, "data is required when ActionResult is a success.");
+			assert(build.projectDirectory, "projectDirectory is required when ActionResult is a success.");
+			assert(build.projectType, "projectType is required when ActionResult is a success.");
+		}
+		if (build.status === ActionResult.ERROR) {
+			assert(build.errorMessages, "errorMessages is required when ActionResult is an error.");
+		}
+	}
+
+	get projectType() {
+		return this._projectType;
+	}
+
+	get projectDirectory() {
+		return this._projectDirectory;
+	}
+
+	get includeUnitTesting() {
+		return this._includeUnitTesting;
+	}
+
+	get npmInstallSuccess() {
+		return this._npmInstallSuccess;
+	}
+
+	static get Builder() {
+		return new class Builder {
+			constructor() { }
+
+			withSuccess() {
+				this.status = ActionResult.SUCCESS;
+				return this;
+			}
+
+			withError(errorMessages) {
```

Comment:
This is a little bit tricky. We name it withError but store errorMessages. Maybe we should rename it.

Reply:
we would rename it to error(errorMessages) this way is compared with the other renaming 
withSuccess() to success()